### PR TITLE
[Py]  (Partially) automatic generation of python API doc

### PR DESCRIPTION
**Pull request by arogowie-intel**
- Detailed documentation is generated automatically
- Summaries are generated automatically for all files except runtime.py
  which is parsed incorrectly.
- No additional steps required. Only run `make html`. Under the hood there
  is lunched script which generates all required files to some temporary
  directory under `/tmp/sphinx_auto_py_doc/` and copied to docs directory
  afterwards.
- Remove `autosummary_generate` from configuration file since we use
  custom script which performs this step.

### CODE REVIEW

File: doc/sphinx/generate_python_api_doc.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+###############################################################################
+# 
+# This script generates stub files for automatically generated documentation.
```
Comment (postrational) [inline doc/sphinx/generate_python_api_doc.sh]: ... of the nGraph Python API.

File: doc/sphinx/generate_python_api_doc.sh [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+DOC_DIR=${NGRAPH_REPO}/doc
+TMP_DIR=/tmp/sphinx_auto_py_doc
+
+
+cd ${NGRAPH_REPO}/python/ngraph
+PYTHONPATH=. sphinx-autogen -t ${DOC_DIR}/sphinx/source/_templates/ -o ${TMP_DIR} ${DOC_DIR}/sphinx/source/python_api/structure.rst
+
+EXCLUDE_DIRS="${NGRAPH_REPO}/python/ngraph/impl* 
+              ${NGRAPH_REPO}/python/ngraph/utils*"
+CURRENT_DIR=.
```
Comment (postrational) [inline doc/sphinx/generate_python_api_doc.sh]: Can we keep all the lines setting vars together at the top?

File: doc/sphinx/generate_python_api_doc.sh [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+# paths relative to this file location
+NGRAPH_REPO=../..
+DOC_DIR=${NGRAPH_REPO}/doc
+TMP_DIR=/tmp/sphinx_auto_py_doc
+
+
+cd ${NGRAPH_REPO}/python/ngraph
+PYTHONPATH=. sphinx-autogen -t ${DOC_DIR}/sphinx/source/_templates/ -o ${TMP_DIR} ${DOC_DIR}/sphinx/source/python_api/structure.rst
+
+EXCLUDE_DIRS="${NGRAPH_REPO}/python/ngraph/impl*
```
Comment (postrational) [inline doc/sphinx/generate_python_api_doc.sh]: Can we keep all the lines setting vars together at the top?

File: doc/sphinx/generate_python_api_doc.sh [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+###############################################################################
+
+# paths relative to this file location
+NGRAPH_REPO=../..
+DOC_DIR=${NGRAPH_REPO}/doc
+TMP_DIR=/tmp/sphinx_auto_py_doc
+
+
+cd ${NGRAPH_REPO}/python/ngraph
+PYTHONPATH=. sphinx-autogen -t ${DOC_DIR}/sphinx/source/_templates/ -o ${TMP_DIR} ${DOC_DIR}/sphinx/source/python_api/structure.rst
```
Comment (postrational) [inline doc/sphinx/generate_python_api_doc.sh]: Can we keep all the lines setting vars together at the top?

File: doc/sphinx/generate_python_api_doc.sh (hunk 1 above)
Comment (arogowie-intel) [reply doc/sphinx/generate_python_api_doc.sh]: Done

File: doc/sphinx/generate_python_api_doc.sh (hunk 2 above)
Comment (arogowie-intel) [reply doc/sphinx/generate_python_api_doc.sh]: Yes.

File: doc/sphinx/generate_python_api_doc.sh (hunk 4 above)
Comment (arogowie-intel) [reply doc/sphinx/generate_python_api_doc.sh]: Here I set PYTHONPATH variable only for this single command `sphinx-autogen`.

**Pull request merged**